testing/shuffle.cu

```diff
@@ -60,27 +352,75 @@ void TestHostDeviceIdentical(size_t m) {
 }
 DECLARE_VARIABLE_UNITTEST(TestHostDeviceIdentical);
 
+template <typename T> void TestFunctionIsBijection(size_t m) {
+  thrust::default_random_engine host_g(0xD5);
+  thrust::default_random_engine device_g(0xD5);
+
+  thrust::system::detail::generic::feistel_bijection host_f(m, host_g);
+  thrust::system::detail::generic::feistel_bijection device_f(m, device_g);
+
+  if (host_f.nearest_power_of_two() >= std::numeric_limits<T>::max() || m == 0) {
+    return;
+  }
+
+  thrust::host_vector<T> host_result(host_f.nearest_power_of_two());
+  thrust::host_vector<T> device_result(device_f.nearest_power_of_two());
+  thrust::sequence(host_result.begin(), host_result.end(), 0llu);
+  thrust::sequence(device_result.begin(), device_result.end(), 0llu);
+
+  thrust::transform(host_result.begin(), host_result.end(), host_result.begin(),
+                    host_f);
+  thrust::transform(device_result.begin(), device_result.end(),
+                    device_result.begin(), device_f);
+
+  ASSERT_EQUAL(host_result, device_result);
+
+  thrust::sort(host_result.begin(), host_result.end());
+  // Assert all values were generated exactly once
+  for (uint64_t i = 0; i < m; i++) {
+    ASSERT_EQUAL((uint64_t)host_result[i], i);
+  }
+}
+DECLARE_VARIABLE_UNITTEST(TestFunctionIsBijection);
+
+void TestBijectionLength() {
+  thrust::default_random_engine g(0xD5);
+
+  uint64_t m = 3;
+  thrust::system::detail::generic::feistel_bijection f(m, g);
+  ASSERT_EQUAL(f.nearest_power_of_two(), 4);
+
+  m = 2;
+  f = thrust::system::detail::generic::feistel_bijection(m, g);
+  ASSERT_EQUAL(f.nearest_power_of_two(), 2);
+
+  m = 0;
+  f = thrust::system::detail::generic::feistel_bijection(m, g);
+  ASSERT_EQUAL(f.nearest_power_of_two(), 1);
+}
+DECLARE_UNITTEST(TestBijectionLength);
+
 // Individual input keys should be permuted to output locations with uniform
 // probability. Perform chi-squared test with confidence 99.9%.
-template <typename Vector>
-void TestShuffleKeyPosition() {
+template <typename Vector> void TestShuffleKeyPosition() {
```

Comment:
Revert

Reply:
There are several more unrelated changes in this PR -- I'll stop pointing them out, but please revert them.

(I have a tentative plan to do a project-wide restyling that will preserve history, so I'd rather not touch lines that don't relate to the current PR topic.)

Reply:
I'm just curious, is there a clang-format or similar file that describes the target style for future reference?
I don't remember seeing anything, though I didn't look too hard.

Reply:
Not yet. The current style of Thrust and CUB is pretty wild and changes from file to file, so it's sort of ill-defined. I have a clang-format file that I use locally, but it's not definitive (and unfortunately, clang-format has no idea what to do with things like `__host__` and `__device__`, so we currently can't rely on it).

New code coming in can be whatever style is reasonable and matches what's already in the file, but I want to hold off on restyling existing code for now to preserve history when we do switch to clang-format.

Reply:
Good to know. Would be great for external contributors to have clang-format eventually.

Reply:
Agreed -- it'd be great for everyone, internal and external! :)